### PR TITLE
Feat: Implement whatsapp with vonage dart

### PR DESCRIPTION
**Pull request by tejas-raskar**
## What does this PR do?

Implements `whatsapp with vonage` using dart

## Test Plan

https://github.com/appwrite/templates/assets/118300989/96f458d3-9e8a-4aaa-94a1-cbdc9ce0631a

## Related PRs and Issues
#200 

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

Yes

### CODE REVIEW

File: dart/whatsapp_with_vonage/lib/main.dart [hunk 1]
```diff
@@ -0,0 +1,69 @@
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:crypto/crypto.dart';
+import 'package:dart_jsonwebtoken/dart_jsonwebtoken.dart';
+import 'package:whatsapp_with_vonage/utils.dart';
+import 'package:http/http.dart' as http;
+
+Future<dynamic> main(final context) async {
+  Map<String, String> env = Platform.environment;
```
Comment (loks0n) [inline dart/whatsapp_with_vonage/lib/main.dart:10]: Can we remove this variable and just use it inline?

File: dart/whatsapp_with_vonage/lib/main.dart [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  throwIfMissing(env, [
+    'VONAGE_API_KEY',
+    'VONAGE_API_SECRET',
+    'VONAGE_API_SIGNATURE_SECRET',
+    'VONAGE_WHATSAPP_NUMBER'
+  ]);
```
Comment (loks0n) [inline dart/whatsapp_with_vonage/lib/main.dart:16]: Can we add an empty newline here?

File: dart/whatsapp_with_vonage/lib/main.dart [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+  }
+
+  final jwt = JWT.verify(token, SecretKey(apiSecretKey!));
+
+  if (jwt.payload['payload_hash'] == null) {
+    return context.res.send('Missing payload hash', 400);
+  }
+
+  final payloadHash =
+      sha256.convert(utf8.encode(context.req.bodyRaw)).toString();
```
Comment (loks0n) [inline dart/whatsapp_with_vonage/lib/main.dart:44]: Formatting looks odd... are you using dart fmt?

**Review by loks0n — CHANGES_REQUESTED** [review]

File: dart/whatsapp_with_vonage/lib/main.dart (hunk 3 above)
Comment (tejas-raskar) [reply dart/whatsapp_with_vonage/lib/main.dart]: yeah, ran the dart format command from the cli

**Review by loks0n — CHANGES_REQUESTED** [review]

**Comment by tejas-raskar** [comment]
Hey @loks0n, I have made the necessary changes. Please review

**Review by loks0n — CHANGES_REQUESTED** [review]

**Comment by tejas-raskar** [comment]
Hey @loks0n, changed the error handling to use JSON now. 
Please review

File: dart/whatsapp_with_vonage/lib/utils.dart [hunk 4]
```diff
@@ -0,0 +1,31 @@
+import 'dart:io';
+import 'package:path/path.dart' as p;
+
+final staticFolder = p.join(p.dirname(Platform.script.toFilePath()), '../static');
+
+/// Throws an error if any of the keys are missing from the object
+/// Parameters:
+///   obj - The object to check
+///   keys - The list of keys to check for
+/// throws Exception
```
Comment (loks0n) [inline dart/whatsapp_with_vonage/lib/utils.dart:10]:
```dart
/// Throws an error if any of the keys are missing from the object
/// @param obj - The object to check
/// @param keys - The list of keys to check for
/// @throws Exception
```

In other templates, we use this format - can we match it here?

File: dart/whatsapp_with_vonage/lib/utils.dart [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+  if (missing.isNotEmpty) {
+    throw StateError('Missing environment variables: ${missing.join(', ')}');
+  }
+}
+
+/// Returns the contents of a file in the static folder
+/// Parameters:
+///    fileName - The name of the file to read
+/// returns Contents of static/{fileName}
```
Comment (loks0n) [inline dart/whatsapp_with_vonage/lib/utils.dart:27]:
```dart
/// Throws an error if any of the keys are missing from the object
/// @param obj - The object to check
/// @param keys - The list of keys to check for
/// @throws Exception
```

In other templates, we use this format - can we match it here?

**Review by loks0n — CHANGES_REQUESTED** [review]
Small final change before merge :)

Don't worry about the hacktoberfest deadline - we will add the `hacktoberfest-accepted` label before it arrives.

File: dart/whatsapp_with_vonage/lib/utils.dart (hunk 4 above)
Comment (tejas-raskar) [reply dart/whatsapp_with_vonage/lib/utils.dart]: Updated

File: dart/whatsapp_with_vonage/lib/utils.dart (hunk 5 above)
Comment (tejas-raskar) [reply dart/whatsapp_with_vonage/lib/utils.dart]: Updated

**Comment by tejas-raskar** [comment]
Hey @loks0n,
Can we merge the PR before hacktoberfest so I can be eligible for the Appwrite swags?

**Comment by gewenyu99** [comment]
Hey there! There were a lot of big PRs during this Hacktoberfest, and we wanted to give everyone ample time to collaborate with our engineering team. If you were able to merge your PRs during October, amazing. If it’s still not merged, don’t worry about it either. Either way, we’ve got your Hacktoberfest swag minted and ready to ship. 

Please comment with your Discord username here so we can contact you about your shipping information to deliver your Hacktoberfest swag.

**Comment by tejas-raskar** [comment]
> Hey there! There were a lot of big PRs during this Hacktoberfest, and we wanted to give everyone ample time to collaborate with our engineering team. If you were able to merge your PRs during October, amazing. If it’s still not merged, don’t worry about it either. Either way, we’ve got your Hacktoberfest swag minted and ready to ship. 
> 
> Please comment with your Discord username here so we can contact you about your shipping information to deliver your Hacktoberfest swag.

Hey @gewenyu99, my discord username is tjsrskr

**Comment by gewenyu99** [comment]
Be in touch soon!

**Pull request merged**